src/api/viewportFeaturesBinary.js

```diff
@@ -0,0 +1,133 @@
+import bboxPolygon from '@turf/bbox-polygon';
+import intersects from '@turf/boolean-intersects';
+import { prepareViewport, isTileFullVisible } from './viewportFeatures';
+
+const GEOMETRY_TYPES = Object.freeze({
+  Point: 0,
+  LineString: 1,
+  Polygon: 2
+});
+
+function addIntersectedFeaturesInTile({ map, data, viewportIntersection, type }) {
+  const indices = getIndices(data);
+  const { positions } = data;
+
+  for (let i = 0; i < indices.length - 1; i++) {
+    const startIndex = indices[i];
+    const endIndex = indices[i + 1];
+    const featureId = getFeatureId(data, startIndex);
+
+    if (!map.has(featureId)) {
+      const ringCoordinates = [];
+
+      for (let j = startIndex; j < endIndex; j++) {
+        ringCoordinates.push(
+          Array.from(
+            positions.value.subarray(j * positions.size, (j + 1) * positions.size)
+          )
+        );
+      }
+
+      if (intersects(getFeatureByType(ringCoordinates, type), viewportIntersection)) {
+        map.set(featureId, getPropertiesFromTile(data, startIndex));
+      }
+    }
+  }
+}
+
+function getIndices(data) {
+  const indices = data.primitivePolygonIndices || data.pathIndices || data.featureIds;
+  return indices.value;
+}
+
+function getFeatureId(data, startIndex) {
+  return data.featureIds.value[startIndex];
+}
+
+function getPropertiesFromTile(data, startIndex) {
```

Comment:
getClonedPropertiesFromTile ?

Reply:
No, we are getting properties from the tile. Cloned is because we need to mix binary and not binary data. It is something that is part of the logic of the function and should not be part of its name because it does not provide information about what the method does